### ISSUE TITLE
Capacitor code not injected to external websites when using * in allowNavigation

### ISSUE THREAD

**Issue by jcesarmobile** [ionic-team/capacitor#1389]
Description of the problem:
When using * in allowNavigation, the Capacitor javascript is not injected to the external page
Create an app that just redirects to https://www.google.es (or your country one) and add
```
"server": {
    "allowNavigation": [
      "*.google.es"
    ]
  }
```
Check Capacitor object and it won't be there.

Change to

```
"server": {
    "allowNavigation": [
      "www.google.es"
    ]
  }
```
and Capacitor object should be there

Affected platform
- [x] Android
- [ ] iOS
- [ ] electron
- [ ] web

OS of the development machine
- [x] Windows
- [x] macOS
- [x] linux

Other information:

Capacitor version: beta 19

node version:

npm version:

CocoaPods version:

Steps to reproduce:

Link to sample project:

**Comment by leshik** [comment]
@jcesarmobile I did some tests on the fresh project and here is what I have found (let's use `www.example.com` for consistency and reproducibility; it features `https` too):

1. Wildcards aren't working as you already mentioned in this issue;
1. When the domain match exactly the invoked URL, it works (e.g. you put `www.example.com` to `allowNavigation`, then open `https://www.example.com` in WebView);
1. If the domain match, but the URL has some path, `Capacitor` object is not injected. For example:

`capacitor.config.json`:
```json
{
  "appId": "com.example.app",
  "appName": "App",
  "bundledWebRuntime": true,
  "webDir": "www",
  "server": {
    "allowNavigation": [
      "www.example.com"
    ]
  }
}
```

`www/index.html`:
```html
<!DOCTYPE html>
<html>
<head>
  <script>
    location.href = 'https://www.example.com/test'
  </script>
</head>
<body>
</body>
</html>
```

results in `Uncaught ReferenceError: Capacitor is not defined` error.

Plain `http` isn't working in this case either, this error isn't related to `https`.

**Comment by Venryx** [comment]
Just wanted to mention that, although the pull-request adding wildcard support is very helpful, there is an edge-case: the URI matcher does not consider the port segment to be a separate segment.

Example: For the URI `192.168.0.31:4001`, it counts `31:4001` as one segment. (ie. it splits only on the `.` character, not `:`)

So to match it, you must use `192.168.*.*`, rather than `192.168.*.*:4001` or `192.168.*.*:*`.

----

What was confusing me is that I tried all of these:
1) `192.168.0.31` -> fails, since `31:4001` doesn't match `31`.
2) `192.168.*.*:4001` -> fails, since `31:4001` doesn't match `*:4001`. (wildcard not recognized)
2) `192.168.*.*:*` -> fails, since `31:4001` doesn't match `*:*`. (wildcard not recognized)

However, these work just fine:
1) `192.168.0.31:4001`
2) `192.168.*.*`

In other words, a "non-wildcard" 4th-digit requires that you enter the port, whereas a "wildcard" 4th-digit requires that you do _not_ enter the port. (which is a bit confusing)

Real-world drawback: It forces you to either...
1) Hard-code what ip-address all developers must have their `webpack-dev-server` on: `192.168.0.31:4001`
2) Remove the port requirement, to fix recognition of the wildcard: `192.168.*.*`

The second is obviously better -- though it is a bit unfortunate that you can't restrict it to a specific port, if using a wildcard for the last ip-address segment.
